include/settings/settings.h

```diff
@@ -93,6 +96,18 @@ struct settings_handler {
 	 */
 };
 
+#if defined(CONFIG_SETTINGS_ROM_HANDLERS)
```

Comment:
Afaik #if defined are not recommended on header since that means the documentation should also be conditional.

Reply:
I think I can safely remove all of them in settings.h, with the disadvantage of using a little more space for the static handlers.